aries_cloudagent/protocols/actionmenu/v1_0/base_service.py

```diff
@@ -28,19 +29,24 @@ async def get_instance(context: InjectionContext):
 
     @abstractmethod
     async def get_active_menu(
-        self, connection: ConnRecord = None, thread_id: str = None
+        self,
+        profile: Profile,
+        connection: ConnRecord = None,
+        thread_id: str = None,
     ) -> Menu:
         """
         Render the current menu.
 
         Args:
+            profile: The profile
             connection: The active connection record
             thread_id: The thread identifier from the requesting message.
         """
 
     @abstractmethod
     async def perform_menu_action(
         self,
+        profile: Profile,
```

Comment:
Not sure if this is the best way to get access to the profile for the event.notify.